inc/saiswitch.h

```diff
@@ -3000,6 +3003,17 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_AVAILABLE_SYSTEM_VOQS,
 
+    /**
+     * @brief Default SAI POE ID
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_POE_DEVICE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_SWITCH_ATTR_POE_DEVICE_ID,
```

Comment:
Looks like it's proposed to have one-to-one mapping of SAI_OBJECT_TYPE_POE_DEVICE to SAI_OBJECT_TYPE_SWITCH, correct?

Reply:
Looking at it again, perhaps SAI_OBJECT_TYPE_POE_DEVICE should not be mapped to SAI_OBJECT_TYPE_SWITCH at all. Since PoE has its own software and hardware that is separated.

Reply:
In general, other use-cases (like PHY or Fabric/VoQ) use a switch object for slightly different reasons. E.g., PHY use case considers PHY as L1 Switch object which makes sense. From what I understand, that's not the case with PoE which uses switch object just as a container or as an entry point for PoE specific syncd application instance to create PoE abstraction.

Reply:
Thank you!

Reply:
> Looking at it again, perhaps SAI_OBJECT_TYPE_POE_DEVICE should not be mapped to SAI_OBJECT_TYPE_SWITCH at all. Since PoE has its own software and hardware that is separated.

Yes. The PoE controller HW is entirely different from switch silicon. The switch (MAC device) and the PoE controllers are entirely different HW elements. The device Id for PoE controllers is different from the MAC/PHY device Id/type.